src/rgw/rgw_sync.cc

```diff
@@ -1616,19 +1660,25 @@ class RGWMetaSyncShardCR : public RGWCoroutine {
       total_entries = sync_marker.pos;
 
       /* sync! */
+      work_period_end = ceph::coarse_mono_clock::now() + work_duration;
       do {
-        if (!lease_cr->is_locked()) {
-          tn->log(10, "lost lease");
-          lost_lock = true;
-          break;
+      while (!rebiddable_lease_cr->is_locked()) {
+        if (rebiddable_lease_cr->is_done()) {
```

Comment:
didn't `full_sync()` already wait on this lock acquisition above on line 1637?

maybe this part inside the do-while loop should just check `if (!rebiddable_lease_cr->is_locked()) {` for an early exit like you do below in `incremental_sync()`?